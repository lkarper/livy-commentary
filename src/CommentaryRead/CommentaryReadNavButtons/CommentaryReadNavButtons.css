.CommentaryReadNavButtons__outer-container {
    position: fixed;
    top: 100px;
    width: 100%;
    left: 0;
    right: 0;
    z-index: 100;
    display: flex;
    flex-direction: column;
    background-color: lightgray;
}

.CommentaryReadNavButtons__mobile-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding: 0 10px;
    width: 100%;
}

.CommentaryReadNavButtons__menu-button {
    font-size: 30px;
}

.CommentaryReadNavButtons__container {
    display: flex;
    flex-direction: row;
    justify-content: center;
    flex-wrap: wrap;    
    width: 100%;
}

.CommentaryReadNavButtons__nav {
    display: flex;
    flex-direction: column;
    z-index: 100;
}

.CommentaryReadNavButtons__nav.hidden {
    display: none;
}

.CommentaryReadNavButtons__button {
    margin: 0;
    padding: 5px;
    border: 1px solid black;
    font-size: 14px;
    text-align: center;
    background-color: blue;
    color: white;
}

.CommentaryReadNavButtons__button.selected {
    color: white;
    background-color: darkblue;
}

@media screen and (min-width: 900px) {
    .CommentaryReadNavButtons__nav.hidden {
        display: flex;
    }

    .CommentaryReadNavButtons__mobile-container {
        display: none;
    }
}