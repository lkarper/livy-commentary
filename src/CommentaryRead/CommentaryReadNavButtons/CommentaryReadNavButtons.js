import React from 'react';
import { NavLink } from 'react-router-dom';
import './CommentaryReadNavButtons.css';

const CommentaryReadNavButtons = (props) => {

    const { book, search } = props;

    return (
        <nav className='CommentaryReadNavButtons__nav'>
            <div className='CommentaryReadNavButtons__container'>
                <NavLink
                    className={`CommentaryReadNavButtons__button ${search.includes(book.book_number) && 'selected'}`}
                    to={`/commentary-read/${book.book_number}`}
                >
                    Book {book.book_number}
                </NavLink>
            </div>
            <div className='CommentaryReadNavButtons__container'>
                {book.chapters.map(chapter => 
                    <NavLink
                        className={`CommentaryReadNavButtons__button ${search.includes(chapter.chapter_number) && 'selected'}`}
                        key={chapter.chapter_number}
                        to={`/commentary-read/${chapter.chapter_number}`}
                    >
                        Chapter {chapter.chapter_number.split('-')[1]}
                    </NavLink>
                    )
                }
            </div>
            <div className='CommentaryReadNavButtons__container'>
                {book.chapters.map(chapter => {
                    return chapter.sections.map(section => 
                        <NavLink
                            className={`CommentaryReadNavButtons__button ${search.includes(section.section_number) && 'selected'}`}
                            key={`${chapter.chapter_number}.${section.section_number}`}
                            to={`/commentary-read/${section.section_number}`}
                        >
                            {chapter.chapter_number.split('-')[1]}.{section.section_number.split('-')[2]}
                        </NavLink>
                        );
                }).flat()}
            </div>
        </nav>
    );
}

export default CommentaryReadNavButtons;