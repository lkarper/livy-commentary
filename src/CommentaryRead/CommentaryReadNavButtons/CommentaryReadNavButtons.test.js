import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CommentaryReadNavButtons from './CommentaryReadNavButtons';

describe('CommentaryReadNavButtons component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <CommentaryReadNavButtons />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <CommentaryReadNavButtons />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
