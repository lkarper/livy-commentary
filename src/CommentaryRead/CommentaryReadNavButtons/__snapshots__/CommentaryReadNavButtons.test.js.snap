// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentaryReadNavButtons component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CommentaryReadNavButtons
      book={
        Object {
          "book_number": 0,
          "chapters": Array [],
        }
      }
      bookNumber=""
      chapterNumber=""
      search=""
      sectionNumber=""
    >
      <div
        className="CommentaryReadNavButtons__outer-container"
      >
        <p>
          Error: Looks like something went wrong; check your connection and try again.
        </p>
      </div>
    </CommentaryReadNavButtons>
  </Router>
</BrowserRouter>
`;
